### ISSUE TITLE
NPE caused by a template - shouldn't happen

### ISSUE THREAD

**Issue by vsajip** [freemarker/freemarker3#1]
The following template:
```
[#macro stmt name value indent]
[#var s = ""?right_pad(indent)]
${s}${name} = ${value} # indented ${indent}
[/#macro]
[@stmt "var1" "value1" 0/]
```
when executed via the command-line, produces, as expected, the result
```
var1 = value1 # indented 0
```
However, if you accidentally omit  the `d` at the end of `right_pad`, as I did, then you get the NPE:
```
java.lang.NullPointerException
	at freemarker.core.ast.BuiltInExpression.isLiteral(BuiltInExpression.java:185)
	at freemarker.core.ast.Expression.setLocation(Expression.java:62)
	at freemarker.core.parser.TemplateLocation.setLocation(TemplateLocation.java:166)
	at freemarker.core.parser.FMParser.BuiltIn(FMParser.java:1565)
	at freemarker.core.parser.FMParser.AddSubExpression(FMParser.java:1447)
	at freemarker.core.parser.FMParser.PrimaryExpression(FMParser.java:483)
	at freemarker.core.parser.FMParser.UnaryExpression(FMParser.java:572)
	at freemarker.core.parser.FMParser.MultiplicativeExpression(FMParser.java:788)
	at freemarker.core.parser.FMParser.AdditiveExpression(FMParser.java:698)
	at freemarker.core.parser.FMParser.RangeExpression(FMParser.java:1028)
	at freemarker.core.parser.FMParser.RelationalExpression(FMParser.java:944)
	at freemarker.core.parser.FMParser.EqualityExpression(FMParser.java:874)
	at freemarker.core.parser.FMParser.AndExpression(FMParser.java:1091)
	at freemarker.core.parser.FMParser.OrExpression(FMParser.java:1151)
	at freemarker.core.parser.FMParser.Exp(FMParser.java:295)
	at freemarker.core.parser.FMParser.Var(FMParser.java:4722)
	at freemarker.core.parser.FMParser.FreemarkerDirective(FMParser.java:4986)
	at freemarker.core.parser.FMParser.Content(FMParser.java:5290)
	at freemarker.core.parser.FMParser.OptionalBlock(FMParser.java:5443)
	at freemarker.core.parser.FMParser.Macro(FMParser.java:3483)
	at freemarker.core.parser.FMParser.FreemarkerDirective(FMParser.java:4876)
	at freemarker.core.parser.FMParser.Content(FMParser.java:5290)
	at freemarker.core.parser.FMParser.OptionalBlock(FMParser.java:5443)
	at freemarker.core.parser.FMParser.Root(FMParser.java:5650)
	at freemarker.template.Template.<init>(Template.java:159)
	at freemarker.cache.TemplateCache.createTemplate(TemplateCache.java:609)
	at freemarker.cache.TemplateCache.loadTemplate(TemplateCache.java:402)
	at freemarker.cache.TemplateCache.getTemplate(TemplateCache.java:335)
	at freemarker.cache.TemplateCache.getTemplate(TemplateCache.java:214)
	at freemarker.template.Configuration.getTemplate(Configuration.java:543)
	at freemarker.template.Configuration.getTemplate(Configuration.java:508)
	at freemarker.core.CommandLine.processTemplate(CommandLine.java:66)
	at freemarker.core.CommandLine.main(CommandLine.java:50)
```
This happened with the version of FreeMarker bundled with JavaCC21.

**Comment by revusky** [comment]
Ah yeah, thanks for that. If you used a non-existent built-in, it should throw a proper ParseException leading to a decent error message with stack trace, but it was just hitting an NPE.  That should be fixed now.

**Comment by vsajip** [comment]
Should be fixed by abbf943.
